### DISCUSSION TITLE
TextPresenter - Colored Text

### DISCUSSION THREAD

**Discussion by drei97** [AvaloniaUI/Avalonia#16586] [Q&A]
Hello,

I want to show some colored dynamic text in my mainwindow, so I am using a TextPresenter control.

The property Foreground sets the text color, but for the whole content. I need different colors in one line, so the first part/word of a line should be red, the second yellow, and so on.

It seems that this is not supported (yet) for TextPresenter. How to do it or are there alternatives to TextPresenter for the purpose described here?

Joachim

**Comment by timunie** [comment]
TextBlock with runs. TextPresenter is only for control templates
